activerecord/lib/active_record/relation.rb

```diff
@@ -432,28 +432,44 @@ def update(id = :all, attributes)
     # Destroys the records by instantiating each
     # record and calling its {#destroy}[rdoc-ref:Persistence#destroy] method.
     # Each object's callbacks are executed (including <tt>:dependent</tt> association options).
-    # Returns the collection of objects that were destroyed; each will be frozen, to
-    # reflect that no changes should be made (since they can't be persisted).
     #
-    # Note: Instantiation, callback execution, and deletion of each
-    # record can be time consuming when you're removing many records at
-    # once. It generates at least one SQL +DELETE+ query per record (or
-    # possibly more, to enforce your callbacks). If you want to delete many
-    # rows quickly, without concern for their associations or callbacks, use
-    # #delete_all instead.
+    # Returns the collection of objects that were destroyed if
+    # +config.active_resource.destroy_all_in_batches+ is set to +false+. Each
+    # will be frozen, to reflect that no changes should be made (since they
+    # can't be persisted).
+    #
+    # Note: Instantiation, callback execution, and deletion of each record can
+    # be time consuming when you're removing many records at once. If you want
+    # to delete many rows quickly, without concern for their associations or
+    # callbacks, use #delete_all instead.
     #
     # ==== Examples
     #
     #   Person.where(age: 0..18).destroy_all
-    def destroy_all(conditions = nil)
-      if conditions
+    #   Person.where(age: 0..18).destroy_all(batch_size: 2000)
+    def destroy_all(conditions_or_options = {})
+      batch_options = conditions_or_options.except!(:begin_at, :end_at, :batch_size)
```

Comment:
I think you intended to use `extract!` instead of `except!`

`except!` removes the keys from given hash and returns that hash
https://github.com/rails/rails/blob/master/activesupport/lib/active_support/core_ext/hash/except.rb#L16

`extract!` removes the keys from given hash and returns hash with that keys
https://github.com/rails/rails/blob/master/activesupport/lib/active_support/core_ext/hash/slice.rb#L44